src/components/cylc/table/deltas.js

```diff
@@ -0,0 +1,215 @@
+          error,
+          job,
+          added,
+          table,
+          lookup
+        ])
+      }
+    }
+  }
+  return result
+}
+
+/**
+ * Deltas updated.
+ *
+ * @param {DeltasUpdated} updated
+ * @param {Table} table
+ * @param {Lookup} lookup - Not really used since we can fetch the data from the table, and we are not altering the lookup data
+ * @returns {Result}
+ */
+function applyDeltasUpdated (updated, table, lookup) {
+  const result = {
+    errors: []
+  }
+  if (updated.taskProxies) {
+    for (const taskProxy of updated.taskProxies) {
+      if (table[taskProxy.id]) {
+        mergeWith(table[taskProxy.id].node, taskProxy, mergeWithCustomizer)
+      }
+    }
+  }
+  if (updated.jobs) {
+    for (const job of updated.jobs) {
+      if (job.firstParent && job.firstParent.id) {
+        const existingTask = table[job.firstParent.id]
+        if (existingTask && existingTask.latestJob.id === job.id) {
+          mergeWith(existingTask.latestJob, job, mergeWithCustomizer)
+        }
+      }
+    }
+  }
+  return result
+}
+
+/**
+ * Deltas pruned.
+ *
+ * @param {DeltasPruned} pruned
+ * @param {Table} table
+ * @param {Lookup} lookup
+ * @returns {Result}
+ */
+function applyDeltasPruned (pruned, table, lookup) {
+  const result = {
+    errors: []
+  }
+  if (pruned.taskProxies) {
+    pruned.taskProxies.forEach(taskProxyId => {
+      Vue.delete(table, taskProxyId)
+    })
+  }
+  if (pruned.jobs) {
+    pruned.jobs.forEach(jobId => {
+      // TODO: should we use an internal lookup for table too? To replace this loop by a quick O(1) operation
+      //       to fetch the existing job with its ID, and then its existing parent task?
+      const parentTask = Object.values(table).find(entry => entry.latestJob && entry.latestJob.id === jobId)
+      if (parentTask && parentTask.latestJob) {
+        Vue.set(parentTask, 'latestJob', {})
+      }
+    })
+  }
+  return result
+}
+
+const DELTAS = {
+  added: applyDeltasAdded,
+  updated: applyDeltasUpdated,
+  pruned: applyDeltasPruned
+}
+
+/**
+ * Handle the deltas for the table view.
+ *
+ * It will use the tasks and jobs of the deltas.
+ *
+ * @param deltas
+ * @param table
+ * @param {Lookup} lookup
+ */
+function handleDeltas (deltas, table, lookup) {
+  const errors = []
+  Object.keys(DELTAS).forEach(key => {
+    if (deltas[key]) {
+      const handlingFunction = DELTAS[key]
+      const result = handlingFunction(deltas[key], table, lookup)
+      errors.push(...result.errors)
+    }
+  })
+  return {
+    errors
+  }
+}
+
+/**
+ * @param {GraphQLResponseData} data
+ * @param {Table} table
+ * @param {Lookup} lookup
+ */
+export default function (data, table, lookup) {
+  const deltas = data.deltas
+  // first we check whether it is a new start
+  if (deltas && deltas.added && deltas.added.workflow) {
+    if (deltas.added.workflow.status === WorkflowState.RUNNING.name) {
```

Comment:
:exclamation:  Note to self: this  is wrong, but will update the PR only after we have decided which direction to go with the table view.

See https://github.com/cylc/cylc-ui/pull/729 and https://github.com/cylc/cylc-ui/issues/667 for more.